olive/engine/engine.py

```diff
@@ -85,19 +85,21 @@ def __init__(
         if execution_providers is None:
             execution_providers = self._config.execution_providers
 
-        # verify the AzureML system have specified the execution providers
-        # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
-        if self.target.system_type == SystemType.AzureML and execution_providers is None:
-            raise ValueError("AzureMLSystem requires execution providers to be specified.")
-        elif execution_providers is None:
-            if self.target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
+        if not execution_providers:
+            if self.target.olive_managed_env:
```

Comment:
I had a same question in Jambay's PR: is another main idea of `olive_managed_env` that user doesn't need to worry about ep and we will try all possible eps for them? I think that was why we skip `get_supported_execution_providers` call for `olive_managed_env`. Or this is a part of Olive auto optimization?

Reply:
yes, for `olive_managed_env` we will create intermediate system and install all the needed ORT.

Reply:
I mean in this case, should we try all possible eps by device? Does user need to provide eps he wants?

Reply:
user need specify what's the EP he/she want to run against multiple EP. and there is no need to try all possible EP